Code/GraphMol/Resonance.cpp

```diff
@@ -1189,26 +1189,37 @@ void ResonanceMolSupplier::mainLoop(unsigned int ti, unsigned int nt) {
 // index -1
 void ResonanceMolSupplier::assignConjGrpIdx() {
   unsigned int nb = d_mol->getNumBonds();
+  if (!nb) return;
+  std::stack<unsigned int> bondIndexStack;
   d_bondConjGrpIdx.resize(nb, -1);
   unsigned int na = d_mol->getNumAtoms();
   d_atomConjGrpIdx.resize(na, -1);
-  for (unsigned int i = 0; i < nb; ++i) {
-    const Bond *bi = d_mol->getBondWithIdx(i);
-    unsigned int biBeginIdx = bi->getBeginAtomIdx();
-    unsigned int biEndIdx = bi->getEndAtomIdx();
-    if (bi->getIsConjugated() && (d_bondConjGrpIdx[i] == -1)) {
-      // assign this conjugate bond to the matching group, if any
-      for (unsigned int j = 0; (d_bondConjGrpIdx[i] == -1) && (j < nb); ++j) {
-        if ((i == j) || (d_bondConjGrpIdx[j] == -1)) continue;
-        const Bond *bj = d_mol->getBondWithIdx(j);
-        if ((bj->getBeginAtomIdx() == biBeginIdx) ||
-            (bj->getBeginAtomIdx() == biEndIdx) ||
-            (bj->getEndAtomIdx() == biBeginIdx) ||
-            (bj->getEndAtomIdx() == biEndIdx))
-          d_bondConjGrpIdx[i] = d_bondConjGrpIdx[j];
+  for (d_nConjGrp = 0; true; ++d_nConjGrp) {
+    for (unsigned int i = 0; i < nb && bondIndexStack.empty(); ++i) {
```

Comment:
Seems like this loop would be clearer as:
```suggestion
    for (const auto bond : d_mol->bonds() ) {
      if(bond->getIsConjugated() && d_bondConjGrpIdx[bond->getIdx()]==-1){
          bondIndexStack.push(bond->getIdx());
          break;
       }
    }
```

Reply:
@greglandrum Done.